__tests__/app.test.js

```diff
@@ -34,111 +34,167 @@ describe("GET /api/topics", () => {
+      return request(app)
+         .get(`/api/articles/${ARTICLE_ID}`)
+         .expect(400)
+         .then((response) => {
+            expect(response.body.msg).toBe(
+               "Please type/select a number to choose an article"
+            );
+         });
+   });
+});
+describe("8. GET /api/articles", () => {
+   test("200: responds with an array of articles sorted by date in descending order", () => {
+      return request(app)
+         .get("/api/articles")
+         .expect(200)
+         .then((response) => {
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
+            expect(response.body).toBeSortedBy('created_at', {descending: true,})
+            console.log(response.body, '<<< my response.body')
+            response.body.forEach((article) => {
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: expect.any(String),
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    comment_count: expect.any(Number),
+                 });
             })
-        })
-    })
-    test('404: responds with a error message when article_id doesn\'t exist', () => {
-        const ARTICLE_ID = 88;
-        return request(app)
-        .get(`/api/articles/${ARTICLE_ID}`)
-        .expect(404)
-        .then((response) => {
-            expect(response.body.msg).toBe('Article not found! Please try again')
-        })
-    })
-    test('400: responds with a error message when client pass article_id which is not a number', () => {
-        const ARTICLE_ID = true;
-        return request(app)
-        .get(`/api/articles/${ARTICLE_ID}`)
-        .expect(400)
+        });
+   });
+   test("200: filter objects by topic given", () => {
+    return request(app)
+        .get('/api/articles?topic=cats')
+        .expect(200)
         .then((response) => {
-            expect(response.body.msg).toBe('Please type/select a number to choose an article')
+            expect(typeof response.body).toBe('object')
+            expect(Array.isArray(response.body)).toBe(true);
+            expect(response.body.length > 0).toBe(true);
+            response.body.forEach((article) => {
+                const TOPIC = 'cats';
+                expect(article).toMatchObject({
+                    author: expect.any(String),
+                    title: expect.any(String),
+                    article_id: expect.any(Number),
+                    topic: TOPIC,
+                    created_at: expect.any(String),
+                    votes: expect.any(Number),
+                    comment_count: expect.any(Number),
+                 });
+            })
+                        
         })
-    })
+   })
+   
```

Comment:
I'd like to see a test for a topic it cannot find (a topic which does not exist in the database) - 404 error

Also how about a test for an invalid topic, code 400.

Also how about a topic which exists, but has no articles attached to it, what would we want that test to look like

Reply:
done